src/utils/sops-utils.ts

```diff
@@ -0,0 +1,105 @@
+import { encryptConfig } from 'api/oauth';
+import { isPlainObject } from 'lodash';
+import { createJSONFormDefaults } from 'services/ajv';
+import { JsonFormsData, Schema } from 'types';
+
+// TODO (typing): Consider adding a type annotation for the promise returned by
+//   the invokeSupabase() function (i.e., src/services/supabase.ts).
+type SupabaseInvokeResponse =
+    | { data: null; error: Error }
+    | { data: any; error: null };
+
+const sopsKey = 'sops';
+
+const copyEncryptedEndpointConfig = (
+    endpointConfigTemplate: { [key: string]: any },
+    encryptedEndpointConfig: { [key: string]: any },
+    encryptedSuffix: string
+) => {
+    Object.entries(encryptedEndpointConfig).forEach(([key, value]) => {
+        let truncatedKey = '';
+        const encryptedSuffixIndex = key.lastIndexOf(encryptedSuffix);
+
+        if (encryptedSuffixIndex !== -1) {
+            truncatedKey = key.slice(0, encryptedSuffixIndex);
+        }
+
+        if (isPlainObject(value)) {
+            copyEncryptedEndpointConfig(
+                endpointConfigTemplate[truncatedKey || key],
+                encryptedEndpointConfig[key],
+                encryptedSuffix
+            );
+        } else {
+            endpointConfigTemplate[truncatedKey || key] = value;
+        }
+    });
+};
+
+const parseEncryptedEndpointConfig = (
+    endpointConfig: { [key: string]: any },
+    endpointSchema: Schema
+): JsonFormsData => {
+    const {
+        sops: { encrypted_suffix },
+        ...encryptedEndpointConfig
+    } = endpointConfig;
+
+    const endpointConfigTemplate = createJSONFormDefaults(endpointSchema);
+
+    copyEncryptedEndpointConfig(
+        endpointConfigTemplate.data,
+        encryptedEndpointConfig,
+        encrypted_suffix
+    );
+
+    return endpointConfigTemplate;
+};
+
+export async function encryptEndpointConfig(
+    endpointConfig: { [key: string]: any },
+    endpointSchema: Schema,
+    serverUpdateRequired: boolean,
+    imageConnectorId: string,
+    imageConnectorTagId: string,
+    callFailed: Function
+): Promise<SupabaseInvokeResponse> {
+    const selectedEndpointConfig =
+        serverUpdateRequired && Object.hasOwn(endpointConfig, sopsKey)
+            ? parseEncryptedEndpointConfig(endpointConfig, endpointSchema).data
+            : endpointConfig;
+
+    let encryptedEndpointConfig: SupabaseInvokeResponse = {
+        data: null,
+        error: {
+            name: 'Object Not Reassigned',
+            message:
+                'An object initializing a the response of a Supabase function was not reassigned.',
+        },
```

Comment:
Even though it will be rare that a user will see this... we should move these to the content file. Also, the message feels weird to me. 

I think it should be one or the other:
'An object initializing **a the** response of a Supabase function was not reassigned.',

Reply:
Haha that is definitely a typo in the message.

As far as moving this content to the language file goes, the consumer of the error expects a direct message (i.e., one not abstracted by a message ID) since this is mocking an error produced by the oauth edge function. Let me confirm that it is not possible to abstract this message.